packages/node_modules/@webex/webex-core/src/lib/services/service-catalogs.js

```diff
@@ -41,6 +54,56 @@ export default class ServiceCatalogs {
     this.hosts = [];
   }
 
+  /**
+   * An active, local, and priority mapped record of the current
+   * {@link ServiceCatalog#hosts}.
+   *
+   * @public
+   * @memberof ServiceCatalog
+   * @type {Record<string, string>}
+   */
+  get map() {
+    // Get a list of active, local, and priority-mapped hosts.
+    return this.find({
+      active: true,
+      local: true,
+      priority: true
+    }).reduce(
+      (map, host) => {
+        // Generate a new object to assign the existing map.
+        const hostReference = {};
+
+        // Assign the key:value pair for the service and url.
+        hostReference[host.service] = host.url;
+
+        // Assign the reference to the map and return.
+        return {...map, ...hostReference};
+      }, {}
+    );
+  }
+
+  /**
+   * Removes a collection of {@link ServiceHost} class objects from the
+   * {@link ServiceCatalogs#hosts} array based on the provided
+   * {@link HostFilter}.
+   *
+   * @public
+   * @memberof ServiceCatalogs
+   * @param {HostFilter} filter - The inclusive filter for hosts to remove.
+   * @returns {this}
+   */
+  clear(filter) {
+    // Collect a list of hosts to remove based on the provided filter.
+    const removing = this.find(filter);
+
+    // Remove the hosts from the array.
+    this.hosts = this.hosts.filter(
+      (host) => !removing.includes(host)
+    );
+
+    return this;
```

Comment:
Since this is modifying the hosts array on `this`, I would expect to not return anything. Or, if it has to return something, it should return `this.hosts` to show what was modified.

Reply:
Improved this to return the removed hosts, much like pop() in array or similar removal methods.